sesman/session.c

```diff
@@ -598,28 +598,48 @@ session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
                 auth_set_env(data);
                 if (s->directory != 0)
                 {
-                    if (s->directory[0] != 0)
+                    if (g_strncasecmp("NO", g_cfg->sec.allowed_alternate_shells, 3) != 0)
                     {
-                        g_set_current_dir(s->directory);
+                        if (s->directory[0] != 0)
```

Comment:
Don't need this test - `g_set_current_dir()` is fine being passed "". Alternatively, on line 599, do what is done with `s->program` below on line 616 and use :-

```
if (s->directory != 0 && s->directory[0] != 0)
```

That's a lot more consistent for the reader I think.

BTW - good spot about the directory change being disabled too. I hadn't thought of that.

Reply:
Okay, I pulled the if statement upwards and unified it as you suggested.

I can't take credit for spotting this, I just took it from the original issue description ;)